app/src/common/shared/org/mozilla/vrbrowser/browser/engine/gecko/GeckoEngine.java

```
@@ -0,0 +1,255 @@
+        return mSettings;
+    }
+
+    @NotNull
+    @Override
+    public TrackingProtectionExceptionStorage getTrackingProtectionExceptionStore() {
+        return null;
+    }
+
+    @NotNull
+    @Override
+    public EngineVersion getVersion() {
+        EngineVersion version = EngineVersion.Companion.parse(MOZILLA_VERSION);
+        if (version != null) {
+            return version;
+        }
+
+        throw new IllegalStateException("Could not determine engine version");
+    }
+
+    @Override
+    public void clearData(@NotNull BrowsingData browsingData, @Nullable String host, @NotNull Function0<Unit> onSuccess, @NotNull Function1<? super Throwable, Unit> onError) {
+        long types = (long) browsingData.getTypes();
+        if (host != null) {
+            mSessionStore.getRuntime().getStorageController().clearDataFromHost(host, types).then(aVoid -> {
+                onSuccess.invoke();
+                return null;
+            }, throwable -> {
+                onError.invoke(throwable);
+                return null;
+            });
+
+        } else {
+            mSessionStore.getRuntime().getStorageController().clearData(types).then(aVoid -> {
+                onSuccess.invoke();
+                return null;
+            }, throwable -> {
+                onError.invoke(throwable);
+                return null;
+            });
+        }
+
+    }
+
+    @NotNull
+    @Override
+    public EngineSession createSession(boolean isPrivate) {
+        return new GeckoEngineSession(mContext, mSessionStore.createSession(isPrivate));
+    }
+
+    @NotNull
+    @Override
+    public EngineSessionState createSessionState(@NotNull JSONObject jsonObject) {
+        return GeckoEngineSessionState.fromJSON(jsonObject);
+    }
+
+    @NotNull
+    @Override
+    public EngineView createView(@NotNull Context context, @Nullable AttributeSet attributeSet) {
+        // We don't support Engine views creation
+        return null;
```

Can we throw an exception here to detect it it's called? I don't want the potential nullpointerexception be hidden in another part of AC code

---

That's a good idea.